### ISSUE TITLE
Never suggest `.pnpm/*` paths in auto imports

### ISSUE THREAD

**Issue by andrewbranch** [microsoft/TypeScript#44511]
This was originally in #44259 but I took it out to narrow the scope. You can still get auto imports into `node_modules/.pnpm` for transitive dependencies, which have no symlinks into top-level `node_modules` folders.

Reverting commit: https://github.com/microsoft/TypeScript/pull/44259/commits/8c439fab4b3c1274ed931f4edf84accc5507518a

Fourslash test:

```ts
/// <reference path="../fourslash.ts" />

// @Filename: /project/tsconfig.json
//// { "compilerOptions": { "module": "commonjs" } }

// @Filename: /project/node_modules/.pnpm/@types+react@17.0.7/node_modules/@types/react/index.d.ts
//// import "csstype";
//// export declare function Component(): void;

// @Filename: /project/node_modules/.pnpm/csstype@3.0.8/node_modules/csstype/index.d.ts
//// export interface SvgProperties {}

// @Filename: /project/index.ts
//// [|import SvgProp/**/|]

// @link: /project/node_modules/.pnpm/@types+react@17.0.7/node_modules/@types/react -> /project/node_modules/@types/react
// @link: /project/node_modules/.pnpm/csstype@3.0.8/node_modules/csstype -> /project/node_modules/.pnpm/@types+react@17.0.7/node_modules/csstype

goTo.marker("");
verify.completions({
  isNewIdentifierLocation: true,
  marker: "",
  exact: [],
  preferences: {
    includeCompletionsForImportStatements: true,
    includeCompletionsWithInsertText: true,
    includeCompletionsWithSnippetText: true,
  }
});
```

**Comment by donaldpipowitch** [comment]
This would fix this new auto-import behaviour which was introduced in 4.3, right?

![image](https://user-images.githubusercontent.com/1152805/121322001-e003c280-c90e-11eb-986d-e382ec6ae58a.png)

**Comment by andrewbranch** [comment]
@donaldpipowitch if `@fortawesome/pro-solid-svg-icons` is one of your direct dependencies, that’s already solved by #44259. You can try typescript@next and you should see better results.

**Comment by donaldpipowitch** [comment]
Cool, thanks!
